src/detect-http2.c

```diff
@@ -0,0 +1,178 @@
+    //TODO create a new doc file for HTTP2 keywords
+    sigmatch_table[DETECT_HTTP2_FRAMETYPE].url = DOC_URL DOC_VERSION "/rules/http2-keywords.html#frametype";
+    sigmatch_table[DETECT_HTTP2_FRAMETYPE].Match = NULL;
+    sigmatch_table[DETECT_HTTP2_FRAMETYPE].AppLayerTxMatch = DetectHTTP2frametypeMatch;
+    sigmatch_table[DETECT_HTTP2_FRAMETYPE].Setup = DetectHTTP2frametypeSetup;
+    sigmatch_table[DETECT_HTTP2_FRAMETYPE].Free = DetectHTTP2frametypeFree;
+#ifdef UNITTESTS
+    sigmatch_table[DETECT_HTTP2_FRAMETYPE].RegisterTests = DetectHTTP2RegisterTests;
+#endif
+    //TODO investigate why this is needed
+    DetectAppLayerInspectEngineRegister("http2",
+                                        ALPROTO_HTTP2, SIG_FLAG_TOSERVER, 0,
+                                        DetectEngineInspectHTTP2);
+    DetectAppLayerInspectEngineRegister("http2",
+                                        ALPROTO_HTTP2, SIG_FLAG_TOCLIENT, 0,
+                                        DetectEngineInspectHTTP2);
+
+    g_http2_match_buffer_id = DetectBufferTypeRegister("http2");
+
+    return;
+}
+
+/**
+ * \brief This function is used to match HTTP2 frame type rule option on a transaction with those passed via http2.frametype:
+ *
+ * \retval 0 no match
+ * \retval 1 match
+ */
+static int DetectHTTP2frametypeMatch(DetectEngineThreadCtx *det_ctx,
+                               Flow *f, uint8_t flags, void *state, void *txv, const Signature *s,
+                               const SigMatchCtx *ctx)
+
+{
+    uint8_t *detect = (uint8_t *)ctx;
+
+    int value = rs_http2_tx_get_frametype(txv, flags);
+    if (value < 0) {
+        //no value, no match
+        return 0;
+    }
+    return *detect == value;
+}
+
+static int DetectHTTP2FuncParseFrameType(const char *str, uint8_t *ft)
+{
+    // first parse numeric value
+    if (ByteExtractStringUint8(ft, 10, strlen(str), str) >= 0) {
+        return 1;
+    }
+
+    // it it failed so far, parse string value from enumeration
+    int r = rs_http2_parse_frametype(str);
+    if (r >= 0) {
+        *ft = r;
+        return 1;
+    }
+
+    return 0;
+}
+
+/**
+ * \brief this function is used to attach the parsed http2.frametype data into the current signature
+ *
+ * \param de_ctx pointer to the Detection Engine Context
+ * \param s pointer to the Current Signature
+ * \param str pointer to the user provided http2.frametype options
+ *
+ * \retval 0 on Success
+ * \retval -1 on Failure
+ */
+static int DetectHTTP2frametypeSetup (DetectEngineCtx *de_ctx, Signature *s, const char *str)
+{
+    uint8_t frame_type;
+
+    if (DetectSignatureSetAppProto(s, ALPROTO_HTTP2) != 0)
+        return -1;
+
+    if (!DetectHTTP2FuncParseFrameType(str, &frame_type)) {
+        SCLogError(SC_ERR_INVALID_SIGNATURE,
+                   "Invalid argument \"%s\" supplied to http2.frametype keyword.", str);
+        return -1;
+    }
+
+    //TODO do we need to allocate ? Could we unionize sm->ctx ?
```

Comment:
Thoughts here ? Allocate one byte looks like an overkill

Reply:
Right now it is necessary. I thought there was a ticket about this, can I can't find it.

Reply:
Do you want me to work on it ?

Reply:
Lets make that a separate project. I think http2 will be complex enough by itself.

Reply:
Indeed, that is a separate project.
But do you want me to create a ticket for this and work on it ?

Reply:
Just the ticket for now. I don't think it's high priority atm.

Reply:
https://redmine.openinfosecfoundation.org/issues/3544